### DISCUSSION TITLE
TFT35 with SKR 1.4 running a RepRap firmware - Use internal SD or USB for Gcode

### DISCUSSION THREAD

**Discussion by WL80** [bigtreetech/BIGTREETECH-TouchScreenFirmware#2294] [Q&A]
Hi all!
I'm doing a retrofit to my printer using a SKR 1.4 board, TFT35 V3.1 and a ESP12s for server.
I used a reprap firmware from gloomyandy (firmware-lpc-wifi-3.3.0_11) and a (DuetWiFiServer-esp8266-lpc-1.26-05) for esp-based server. So far so good.
The TFT is connected using serial only, as EXT1/2 are utilized by ESP.
Files uploaded using reprap / duet web interface are visible in the print menu of the TFT.
Files located on SD card or USB stick inserted directly to TFT are not showing though.
What am I missing / doing wrong?
My intention was to use web interface uploads or local media interchangeably / in parallel.
Ideas?
Cheers and Merry X-mas!
WL

**Comment by WL80** [comment]
Update:
I tried different settings for communication to TFT (an entry in the config.g); results are:
M575 P1 **S2** B57600 (default)
- (-) no access to files on SD card and USB located at the TFT35
- (+) working access to files on SD card at the SKR mainboard (so the Duet Server uploads)
- (+) working readings at the TFT 

M575 P1 **S1** B57600
- (+) working access to files on SD card and USB located at the TFT35
- (-) no access to files on SD card at the SKR mainboard (so the Duet Server uploads). 
- (-) bad (not refreshing properly) readings at the TFT

**Comment by osky087** [comment]
I have the same board and screen, they came with the biqu b1 printer. What wifi adapter did you use to get reprap going? I was thinking of getting the ESP01S that Biqu sells to attempt getting reprap going, but dont know if that will work. I could plug the ESP01S adapter to the board or screen.

**Comment by WL80** [comment]
You need an ESP12s-based module or build one based on that chip. Check the relevant repository here on github. Alternatively you can buy a ready board is available in various shops. Just look for "rrf wifi skr 1.4". 
I need the python and the esptool, as the direct programming from SKR did not work for me. Better get a board with built-in USB for easy flash, otherwise you may need a bit soldering / external programming as I did.
**About my question, anyone - what am I doing wrong? Is there an inherent problem I'm not seeing / a logical conflict in how the system should be used?**

**Comment by radek8** [comment]
The TFT35 V3.1 display is not from Bigtreetech ...
Seek support from the manufacturer of the three-trees.

**Reply by WL80** [reply]
It is two trees indeed, the HW is identical or almost identical while actual problem shows as described above. Any practical suggestions?
Cheers/pozdrawiam

**Comment by radek8** [comment]
BTT does not have version 3.1
version 3.1 is TWOTREES

![image](https://user-images.githubusercontent.com/46979052/147572266-8a8e5fba-e964-4f55-83cc-151bf262b9c5.png)

**Comment by WL80** [comment]
What you linked above is an E3 and it is surely different.
V3 and 3.1 I have are practically the same. Microcontroller (stm32f207vct6), layout, terminals, etc. 
3.1 Two Trees:
![TT](https://user-images.githubusercontent.com/96480397/147581474-9f7451dd-415b-4c4d-baaf-a8050912e001.jpg)

3.0 BTT:
![BTT](https://user-images.githubusercontent.com/96480397/147581481-32b6e102-e205-4f68-b1ab-ca59ba43daf1.jpg)

Problem is access being exclusive to either remote SD (at SKR) or local SD/USB depending on the FW setting; all plus described issues. Any ideas?

**Comment by radek8** [comment]
If you connect ESP to SKR logically you can't see the files on SD and USB on TFT
The TFT communicates with the SKR using the G-code and cannot send the contents of the storage to the serial port at the request of the SKR.

**Comment by WL80** [comment]
Idea is to either use SKR file upload and print via ethernet (way Duet would normally do and it seems to work) or use TFT panel locally and storage specific to it (say like typically with Marlin).
In the latter case request would not be from SKR, but local, from TFT loi; a very usual way to print and afaik standard for confing using marlin at SKR.
As shown above, depending on S1 / S2 mode of M575, either seems to work, but not at the same time, with S1 additionally making temperature indication sluggish. Question is how to be able to use both, TFT storage especially.
